sum_up_zero.cpp

```diff
@@ -78,8 +78,9 @@ void print_match(std::set<std::vector<int> > pair_list)
         {
             cout<<*num<<" ";
         }
-        cout<<") "<<endl;
+        cout<<") ";
     }
+	cout<<endl;
 }
 
 
```

Comment:
I was going to write a comment further up but I can only comment on new code :p
Your solution seems very efficient! By using the left and the right place holders you cut down about half the insertions